File: src/main/java/io/papermc/voidWorld/helper/VWDimension.java
package io.papermc.voidWorld.helper;

import org.bukkit.World;

public enum VWDimension {
    OVERWORLD(World.Environment.NORMAL),
    NETHER(World.Environment.NETHER),
    END(World.Environment.THE_END);

    private final World.Environment env;

    VWDimension(World.Environment env) {
        this.env = env;
    }

    public World.Environment getEnvironment() {
        return env;
    }

    public static VWDimension fromString(String name) {
        if (name == null) return null;
        try {
            return VWDimension.valueOf(name.toUpperCase());
        } catch (IllegalArgumentException e) {
            return null;
        }
    }
}

File: src/main/java/io/papermc/voidWorld/helper/VWDimension.kt
package io.papermc.voidWorld.helper

import org.bukkit.World
import java.util.*

enum class VWDimension(val environment: World.Environment) {
    OVERWORLD(World.Environment.NORMAL),
    NETHER(World.Environment.NETHER),
    END(World.Environment.THE_END),
    ;

    companion object {
        @JvmStatic
        fun fromString(name: String?): VWDimension? =
            name
                ?.uppercase(Locale.ROOT)
                ?.let {
                    runCatching { valueOf(it) }.getOrNull()
                }
    }
}
